pkg/app/app_deploy.go

```diff
@@ -146,7 +146,9 @@ func (a *App) trySaveMetadata(kapp *ctldep.Kapp) {
 func (a *App) newKapp(kapp v1alpha1.AppDeployKapp, cancelCh chan struct{}) (*ctldep.Kapp, error) {
 
 	return a.deployFactory.NewKapp(kapp, a.app.Spec.ServiceAccountName,
```

Comment:
this factory is starting to get pretty big for its constructor. I think its fine for this PR, but if we have to add much more - we might want to think about how to make this smaller / a structured input

Reply:
Yep, makes sense.